tket/src/Clifford/include/Clifford/CoherentTableau.hpp

```diff
@@ -0,0 +1,252 @@
+// Copyright 2019-2022 Cambridge Quantum Computing
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "SymplecticTableau.hpp"
+
+namespace tket {
+
+// Forward declare Circuit for friend converter
+class Circuit;
```

Comment:
It would require making all of the internal data of the `CoherentTableau` public (i.e. the underlying `SymplecticTableau` and the map between column indices and qubits). Do we have a standard policy/guidelines wrt encapsulation and hiding?

Reply:
I don't actually see a problem with making these data public. I think we'd only need getters (const methods), so there wouldn't be any scope for client code messing with internal constraints of the class, but I may be missing something.